src/proposal.mligo

```diff
@@ -47,12 +46,40 @@ let ensure_proposing_stage(stage_num, store : nat * storage): storage =
   else (failwith("NOT_PROPOSING_STAGE") : storage)
 
 [@inline]
-let ensure_proposal_is_unique (propose_params, store : propose_params * storage): proposal_key =
-  let proposal_key = to_proposal_key(propose_params, Tezos.sender) in
+let ensure_proposal_is_unique (from_, propose_params, store : address * propose_params * storage): proposal_key =
+  let proposal_key = to_proposal_key(propose_params, from_) in
   if Map.mem proposal_key store.proposals
     then (failwith("PROPOSAL_NOT_UNIQUE") : proposal_key)
     else proposal_key
 
+// -----------------------------------------------------------------
+// Delegate
+// -----------------------------------------------------------------
+
+[@inline]
+let check_delegate (from_, author, store : address * address * storage): address =
```

Comment:
A comment above this function could be useful to tell what it does, for example without looking at the code it's not clear what address is returned.

Reply:
Added a comment on this.